### PR TITLE
Update prefetch docs with browser support information

### PR DESCRIPTION
**Pull request by bluwy**
#### Description (required)

Update the prefetch docs based on the core PR linked below, which includes some internal behaviour changes.

I also added a new browser support section to explain the behaviour difference between browsers, and how to ensure prefetching works reliably.

### CODE REVIEW

File: src/content/docs/en/guides/prefetch.mdx
Comment (bluwy) [inline src/content/docs/en/guides/prefetch.mdx:54]: This will no longer be the case as all strategies will internally use `<link rel="prefetch">` which has the same (low) priority.

File: src/content/docs/en/guides/prefetch.mdx
Comment (bluwy) [inline src/content/docs/en/guides/prefetch.mdx:118]: The `with` option is deprecated so I removed this to discourage people from using it.

File: src/content/docs/en/guides/prefetch.mdx
Comment (sarah11918) [inline src/content/docs/en/guides/prefetch.mdx:163]:
```suggestion
**Support:** Safari does not support `<link rel="prefetch">` and will fall back to the `fetch()` API, which requires cache headers (e.g. [`Cache-Control`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control), [`Expires`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Expires), and [`ETag`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag)) to be set. Otherwise, the prefetch will not work.

**Edge case:** `ETag` headers do not work in private windows.
```

File: src/content/docs/en/guides/prefetch.mdx
Comment (sarah11918) [inline src/content/docs/en/guides/prefetch.mdx:167]:
```suggestion
To best support all browsers, make sure your pages have the proper cache headers. 

For static or prerendered pages, the `ETag` header is often automatically set by the deployment platform, so they should work out of the box.
```

File: src/content/docs/en/guides/prefetch.mdx
Comment (sarah11918) [inline src/content/docs/en/guides/prefetch.mdx:169]:
```suggestion
For dynamic and server-side rendered pages, you should set the appropriate cache headers based on the page content. Visit the [MDN documentation on HTTP caching](https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching) for more information.
```

**Review by sarah11918 — COMMENTED** [review]
Looks great, @bluwy !

I tried something different with the browser support section (haven't seen it in preview yet, so it might be awful). Let's see what we think!

**Comment by bluwy** [comment]
Thanks! I like your suggested formatting too. I committed them and view locally and it looks like this:

<details>
<summary>Screenshot (Vercel link should also be possible)</summary>

image

</details>

Seeing this, maybe we could remove the `Support:` prefix on each paragraphs? Since the heading "Browser support" already implies so.

**Comment by sarah11918** [comment]
Yup, agreed! I had it in my head that maybe the different original list items were going to be different, repeatable "things"... but nope, I think without will be just fine!

**Comment by bluwy** [comment]
Thanks! I've updated and removed them now.

**Comment by bluwy** [comment]
@sarah11918 the core PR has been released. Waiting on your approval just in case and I think we can then merge this

**Comment by sarah11918** [comment]
@bluwy Thanks, taking one more final pass now, then I'll merge!

**Review by sarah11918 — APPROVED** [review]
Thank you, @bluwy , LGTM!

**Pull request merged**
